src/util.ts

```diff
@@ -209,3 +210,15 @@ export function varName(s: string): string {
   // Add _ if the string has leading numbers.
   return (s.match(/^\d+/) ? '_' : '') + alphanumericS;
 }
+
+export function logicalExpr<T>(op: LogicalOperand<T>, cb: Function): string {
+  if (isLogicalNot(op)) {
+    return '!' + logicalExpr(op.not, cb);
```

Comment:
These would be safer with more parentheses `!(${logicalExpr(op.not, cb)})`.

For example, just in case someone apply this to `filter :{not: "A && B"}` then it should become `!(A && B)` rather than `!A && B` -- obviously this is a strawman example but better be safe.
(Note that I imagine logical expression can be applied for `transform.filter` and thus A and B are predicates, not  selection names.)

Similarly, I think we could do `.join(') && (')` instead of `.join(' && ')` (and same for `||`)